plugin/builtin/buildbaron/build_baron.go

```diff
@@ -130,6 +150,44 @@ type searchReturnInfo struct {
 	Search string                  `json:"search"`
 }
 
+// raceSuggesters returns the JIRA ticket results from the altEndpoint suggester if it returns
+// within its configured interval, and returns the JIRA ticket results from the fallback suggester
+// otherwise.
+func raceSuggesters(fallback, altEndpoint suggester, t *task.Task) ([]thirdparty.JiraTicket, error) {
+	type result struct {
+		Tickets []thirdparty.JiraTicket
+		Error   error
+	}
+
+	// thirdparty/jira.go and thirdparty/http.go do not expose an API that accepts a context.Context.
+	fallbackCtx := context.TODO()
```

Comment:
Recommend sticking to just calling these ctx and cancel. This context should also be .Background()

Reply:
why can't we pass context into this?

Reply:
@tychoish, it isn't that we _can't_ but rather that the `context.Context` object would be unused. I have no intention of plumbing `context.Context` through to `thirdparty.JiraHandler` and `thirdparty.httpClient` as part of this pull request.

> TODO returns a non-nil, empty Context. Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter). TODO is recognized by static analysis tools that determine whether Contexts are propagated correctly in a program.
> 
> https://golang.org/pkg/context/#TODO

I had interpreted the documentation to mean that because functions being called wouldn't do anything with the context that `context.TODO` would make this more obvious to existing tools. It is entirely possible that I'm misunderstanding the intent here.

Reply:
Yep, you're right about TODO. No need to plumb the contexts in to the JIRA handler IMO.